### PR TITLE
internal: fix lldb-dap unconditionally calling rustc

### PR DESCRIPTION
**Pull request by davidbarsky**
Fixes https://github.com/rust-lang/rust-analyzer/issues/18318. I also took the opportunity to refactor how `discoverSourceFileMap` worked—it now returns a type instead of mutating a map in place.

I tested this change using the LLDB DAP extension. I needed to set `"lldb-dap.executable-path": "/Applications/Xcode.app/Contents/Developer/usr/bin/lldb-dap"` for everything to work, however, but once I did, was able to successfully debug a test.

### CODE REVIEW

File: editors/code/src/debug.ts [hunk 1]
```diff
@@ -203,9 +221,11 @@ async function discoverSourceFileMap(
         const commitHash = rx.exec(data)?.[1];
         if (commitHash) {
             const rustlib = path.normalize(sysroot + "/lib/rustlib/src/rust");
-            sourceFileMap[`/rustc/${commitHash}/`] = rustlib;
+            return { source: rustlib, destination: rustlib };
         }
     }
+
+    return;
```
Comment (lnicola) [inline editors/code/src/debug.ts:228]: Is this `return` needed?

File: editors/code/src/debug.ts [hunk 2]
```diff
@@ -189,11 +203,15 @@ async function getDebugConfiguration(
     return debugConfig;
 }
 
+type SourceFileMap = {
+    source: string;
+    destination: string;
```
Comment (lnicola) [inline editors/code/src/debug.ts:208]: Nit: maybe `target`, but don't bother changing it.

**Review by lnicola — APPROVED** [review]

File: editors/code/src/debug.ts (hunk 1 above)
Comment (davidbarsky) [reply editors/code/src/debug.ts:228]: afraid so, yeah. Without this return, we'd get a "Not all code paths return a value. `ts(7030)`" error.

**Pull request merged**